src/baseline_bots/bots/smart_order_accepter_bot.py

```diff
@@ -551,13 +552,28 @@ def __call__(self, rcvd_messages: List[Tuple[int, Message]]):
             orders_data.add_orders(best_orders, overwrite=True)
             self.orders = orders_data
 
+
+            # fmt: off
+            # stance is not yet considered in first phase
+            if self.game.get_current_phase()!='S1901M':
+                self.allies = [pow for pow in powers if (pow != self.power_name and powers[pow] >= self.ally_threshold)]
+                self.foes = [pow for pow in powers if (pow != self.power_name and powers[pow] <= self.enemy_threshold)]
+                self.neutral = [pow for pow in powers if (pow != self.power_name and powers[pow] > self.enemy_threshold and powers[pow] < self.ally_threshold)]
+                msg_allies, msg_foes, msg_neutral = ','.join(allies), ','.join(foes), ','.join(neutral)
+                msgs_data.add_message("GLOBAL", str(f"{self.power_name}: From my stance vector perspective, I see {msg_allies if msg_allies else 'no one'} as my allies, \
+                                {msg_foes if msg_foes else 'no one'} as my foes and I am indifferent towards {msg_neutral if msg_neutral else 'no one'}"))
+                yield self.send_message("GLOBAL", str(f"{self.power_name}: From my stance vector perspective, I see {msg_allies if msg_allies else 'no one'} as my allies, \
+                                {msg_foes if msg_foes else 'no one'} as my foes and I am indifferent towards {msg_neutral if msg_neutral else 'no one'}"))
+            # fmt: on
+
+
             # filter out aggressive orders to allies
             yield self.replace_aggressive_order_to_allies()
 
             # generate messages for FCT sharing info orders
             opps = list(powers.keys()).copy()
             opps.remove(self.power_name) # list of opposing powers
-            msgs_data = yield self.gen_messages(orders_data.get_list_of_orders())
+            msgs_data = yield self.gen_messages(orders_data.get_list_of_orders(), msgs_data)
```

Comment:
This is to add more flexibility to `gen_message` It should not create a new msgs_data = MessageData but use the existing one that we already have. Oh though this is (functionally) not in used anymore